### ISSUE TITLE
Investigate Next.js as a replacement for Gatsby

### ISSUE THREAD

**Issue by Chalarangelo** [30-seconds/30-seconds-web#308]
Next.js looks amazing and could suit our needs. Let's investigate, prototype, figure out if it's an option.

Note: Low priority, but huge ROI. Investigation should be conducted in parallel with other tasks and we can switch focus for a Sprint to handle it if it suits us.

**Comment by Chalarangelo** [comment]
I'm moving this up into active development as of today. We can probably run this in parallel with other tasks for the rest of this sprint and the next one.

**Comment by Chalarangelo** [comment]
### Tasks

List may be incomplete

- [x] Bootstrap Next.js
- [ ] Set up scripts and check everything works as expected
- [x] Page metadata via `next/head`
- [x] Move `_redirects` and `robots.txt` to public
- [x] `onPreInit` - Probably not needed, part of the bootstrap process
- [ ] `onPostBuild` - Change mechanisms to write XML and JSON files to public
- [ ] `onCreateWebpackConfig` - Maybe not needed, check docs
- [ ] `onCreateDevServer` - Set up the `/create` route if possible
- [ ] `createPagesStatefully` - Set up a mechanism for this
- [x] `gatsby-browser` - Move to Next.js hook
- [x] `gatsby-node` - Part of the bootstrap process
- [x] `gatsby-ssr` - Part of the `next/head` metadata
- [x] `gatsby-plugin-manifest` - Convert plugin to serializers
- [ ] `gatsby-plugin-catch-links` - Investigate
- [ ] `gatsby-plugin-netlify` - Investigate
- [x] `gatsby-plugin-react-helmet` - Not needed
- [x] `gatsby-plugin-sass` - Not needed
- [x] `gatsby-source-filesystem` - Not needed
- [ ] Remove `react-helmet`
- [ ] Remove `gatsby` dependencies
- [ ] Change credits in footer
- [ ] Check for gatsby leftovers
- [ ] Setup Netlify to point to the build directory (whatever it is), instead of `public`
